### DISCUSSION TITLE
read only mnt

### DISCUSSION THREAD

**Discussion by anthonyfca** [minio/minio#13269] [Q&A] [answered]
Context: I have spent many many yhours over the last three weeks trying and failing to get minio working on my mac, using homebrew as wget doesn't work

unable to use the drive /mnt/data1 mkdir /mnt: read-only file system
unable to use the drive /mnt/data2 mkdir /mnt: read-only file system
unable to use the drive /mnt/data3 mkdir /mnt: read-only file system
unable to use the drive /mnt/data4 mkdir /mnt: read-only file system

I have 4 new 6tb disks connected named DISK1 DISK2 DISK3 DISK4
I have no clue how minio identifies the correct available disks, cannot find any clues anywhere.

## Expected Behavior

My guess? minio install defaults to use of /mnt and Apple has now made all folders off root system read only from some earlier version before big sur.

How do I change this default to say use a folder off my home directory?

## Steps to Reproduce (for bugs)
export MINIO_ROOT_USER=minio
     export MINIO_ROOT_PASSWORD=miniominio
     minio server http://node{1...32}.example.com/mnt/export{1...4}

1.
2.
3.
4.

## Context

Failure. Unable to install.

A set of command lines that will give me a worm version of minio installed space on my 4 disks just purchased for the purpose, without the constant digressions in the docs that make it very difficult if not beyond my power to follow correctly. sadly. I REALLY have tried.

## Regression

Don't know what regression is.

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

version minio version RELEASE.2021-09-18T18-09-59Z

no setup yet
Big sur 11.6 

doing my best here, I am not a developer in these things. 

I followed the slack links but that didn't work for me either, never used slack before.

regards

anthony

**Comment by klauspost** [comment, marked as answer]
`http://node{1...32}.example.com/mnt/export{1...4}` is defining your cluster setup.

`node{1...32}.example.com` tells it that there will be 32 server nodes, named `node1.example.com`,  `node2.example.com`. These will need to be resolved through DNS. But I suspect that is not the setup you want. Changing this to `127.0.0.1:9000` will tell it is a single server.

`mnt/export{1...4}` tells the location of your disks. In this case it will look for `/mnt/export1`, `/mnt/export2`... Change that to the locations of your disks.

**Reply by anthonyfca** [reply]
apologies for my slowness, I only just discovered this  is in discussions and had only the email copy.

this is progress based on your help:

minitv:/ tv$ minio server ~/NODE./Volumes/DISK{1...4}/data
Automatically configured API requests per node based on available memory on the system: 82
Status:         4 Online, 0 Offline. 
API: http://10.10.0.116:9000  http://10.10.0.63:9000  http://127.0.0.1:9000                   
RootUser: minio 
RootPass: miniostorage 

Console: http://10.10.0.116:55364 http://10.10.0.63:55364 http://127.0.0.1:55364          
RootUser: minio 
RootPass: miniostorage 

Command-line: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set myminio http://10.10.0.116:9000 minio miniostorage

Documentation: https://docs.min.io

WARNING: Console endpoint is listening on a dynamic port (55364), please use --console-address ":PORT" to choose a static port.

I cannot figure out how to make that port correction, all I get is this sort of thing:

minitv:/ tv$ minio server --console-127.0.0.1:9000 ~/mnt/data{1...4}
Incorrect Usage: flag provided but not defined: -console-127.0.0.1:9000

I've tried all sorts of variations of syntax to no avail.

Yes I'd prefer not to have to delve in to DNS as it's all local.

wow! I just got the gui up after removing the " . " after NODE
BUT I have not fixed the PORT issue

dare I admit to a certain thrill when the gui came up? 

LATER: SO NOW IT SEEMS 

minio server /Volumes/DISK{1...4}/data

is all I needed. 

which in its shorthand represents
/Volumes/DISK1/data
/Volumes/DISK2/data
/Volumes/DISK3/data
/Volumes/DISK4/data

I do NOT know if the /data directories have to pre-exist - mine did, I think they do.

so had I known at the beginning that all this has been about is a simple (regardless of how complex) pathname, things would have been very nice instead of very painful.

IF (BIG IF) I have now understood this process I could write it up in a way less nebulous for us lesser mortals to grasp. 

In my previous syntax above:
using the command with NODE created a directory in the home director called node and sub directories for the 4 "distributed" locations, so my seeing this and so realising that all NODE is, is a path, I removed it so the command has only the local path and (before testing!) it works and I get the gui up absent the buckets I created in the first attempt above.

PORT 9000 is NOT solved.

anthony

**Reply by klauspost** [reply]
Use `--console-address :9000` or `--console-address 127.0.0.1:9000`.
